### ISSUE TITLE
v1.6.3: Segmentation: validation metric flat at 1.000

### ISSUE THREAD

**Issue by LeMinhNgan** [roboflow/rf-detr#939]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

Hello,
I tried RFDETRSegSmall and see the all the validation metrics are flat at 1.00 even from the first epoch, meanwhile the val/loss is still gradually reducing.
The "valid" folder contain 2 images.
The screenshot is attached herein

P/s: i actually found this issue from version 1.4.x and 1.5.x the mAP reach 0.5 after several epochs and standstill regardless training in progress

### Environment

- rfdetr: 1.6.3
- OS windows11
- RTX 5090
- Python 3.10
- cuda 12.8

### Minimal Reproducible Example

<img width="1920" height="1035" alt="Image" src="https://github.com/user-attachments/assets/621682cb-7fae-4a03-92b7-4253ae9e2838" />

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by Borda** [comment]
@LeMinhNgan Could you share the dataset with us? From the picture you shared sems that the validation dataset is perfect, but still some room for improvement on trainig split of dataset

**Comment by LeMinhNgan** [comment]
I will work around again with the dataset splitting. The result is too perfect to be true, even though I dont care the metrics much, I just look at the train/loss and it seem OK. 
I just let you know it if someone may mention this question in future, that may help you

**Comment by Borda** [comment]
It is possible that there is a bug but we need reproducible example for debugging...
